lib/src/model.dart

```diff
@@ -325,9 +325,12 @@ abstract class ModelElement {
           buf.write(p.modelType.element.linkedParams);
           buf.write(')');
         } else if (p.modelType != null &&
-            p.modelType.element != null &&
-            p.modelType.element.modelType != null) {
-          String typeName = p.modelType.element.modelType.linkedName;
+            p.modelType.element != null) {
```

Comment:
Could you add a test for this fix in `test/fake-package` and `test/model_test.dart`?